packages/vite/src/client/error.ts

```diff
@@ -0,0 +1,266 @@
+import type { RawSourceMap } from 'source-map'
+import { SourceMapConsumer } from 'source-map'
+import type { ErrorPayload } from 'types/hmrPayload'
+
+export const transformError = (error: any): Error => {
+  if (error instanceof Error) {
+    return error
+  }
+
+  const e = Error(error ?? 'unknown')
+  e.stack = `a non-error was thrown please check your browser's devtools for more information`
+  return e
+}
+
+const RE_CHROME_STACKTRACE =
+  /^ {4}at (?:(.+?)\s+)?\(?(.+?)(?::(\d+))?(?::(\d+))?\)?$/
+const RE_FIREFOX_STACKTRACE =
+  /^(?:(?:(^|.+?)@))\(?(.+?)(?::(\d+))?(?::(\d+))?\)?$/
```

Comment:
These feel a bit messy to me but they work. if someone wants to take a crack at making this cleaner i would appreciate it